### PR TITLE
iothread: add case with two iothread inactive

### PR DESCRIPTION
**Pull request by dzhengfy**
This is to test the iothread parameters could be set before vm starts
Case ID: XXX-299032

Depends on 
- https://github.com/avocado-framework/avocado-vt/pull/3791

Signed-off-by: Dan Zheng <dzheng@redhat.com>

### CODE REVIEW

**Comment by dzhengfy** [comment]
libvirt-9.5.0

 (01/22) type_specific.io-github-autotest-libvirt.cpu.iothread.positive_test.iothread_config.only_iothread_id: WARN: Test passed but there were warnings during execution. Check the log for details. (5.52 s)
 (02/22) type_specific.io-github-autotest-libvirt.cpu.iothread.positive_test.iothread_config.only_iothread_num: WARN: Test passed but there were warnings during execution. Check the log for details. (5.63 s)
 (03/22) type_specific.io-github-autotest-libvirt.cpu.iothread.positive_test.iothread_config.both_iothread_id_num: WARN: Test passed but there were warnings during execution. Check the log for details. (5.59 s)
 (04/22) type_specific.io-github-autotest-libvirt.cpu.iothread.positive_test.defaultiothread_config: WARN: Test passed but there were warnings during execution. Check the log for details. (5.94 s)
 (05/22) type_specific.io-github-autotest-libvirt.cpu.iothread.positive_test.thread_pool.shutoff_vm: WARN: Test passed but there were warnings during execution. Check the log for details. (4.81 s)
 (06/22) type_specific.io-github-autotest-libvirt.cpu.iothread.positive_test.thread_pool.live_vm: WARN: Test passed but there were warnings during execution. Check the log for details. (123.54 s)
 (07/22) type_specific.io-github-autotest-libvirt.cpu.iothread.positive_test.iothreadpin: WARN: Test passed but there were warnings during execution. Check the log for details. (93.40 s)
 (08/22) type_specific.io-github-autotest-libvirt.cpu.iothread.positive_test.iothread_poll: WARN: Test passed but there were warnings during execution. Check the log for details. (127.10 s)
 (09/22) type_specific.io-github-autotest-libvirt.cpu.iothread.positive_test.iothread_cold_set: WARN: Test passed but there were warnings during execution. Check the log for details. (49.40 s)
 (10/22) type_specific.io-github-autotest-libvirt.cpu.iothread.positive_test.disk_attach: WARN: Test passed but there were warnings during execution. Check the log for details. (96.23 s)
 (11/22) type_specific.io-github-autotest-libvirt.cpu.iothread.negative_test.iothreadpin.out_of_range: WARN: Test passed but there were warnings during execution. Check the log for details. (5.49 s)
 (12/22) type_specific.io-github-autotest-libvirt.cpu.iothread.negative_test.iothreadpin.no_matchs_iothread: WARN: Test passed but there were warnings during execution. Check the log for details. (4.59 s)
 (13/22) type_specific.io-github-autotest-libvirt.cpu.iothread.negative_test.iothread_poll.invalid_string.poll_max_ns:WARN: Test passed but there were warnings during execution. Check the log for details. (94.36 s)
 (14/22) type_specific.io-github-autotest-libvirt.cpu.iothread.negative_test.iothread_poll.invalid_string.poll_grow: WARN: Test passed but there were warnings during execution. Check the log for details. (95.18 s)
 (15/22) type_specific.io-github-autotest-libvirt.cpu.iothread.negative_test.iothread_poll.invalid_string.poll_shrink:WARN: Test passed but there were warnings during execution. Check the log for details. (93.13 s)
 (16/22) type_specific.io-github-autotest-libvirt.cpu.iothread.negative_test.iothread_poll.no_matching_id: WARN: Test passed but there were warnings during execution. Check the log for details. (92.19 s)
 (17/22) type_specific.io-github-autotest-libvirt.cpu.iothread.negative_test.iothreadsched.no_iothread.no_iothreads_xml: WARN: Test passed but there were warnings during execution. Check the log for details. (4.50 s)
 (18/22) type_specific.io-github-autotest-libvirt.cpu.iothread.negative_test.iothreadsched.no_iothread.no_matching_id:WARN: Test passed but there were warnings during execution. Check the log for details. (4.55 s)
 (19/22) type_specific.io-github-autotest-libvirt.cpu.iothread.negative_test.iothreadsched.invalid: WARN: Test passed but there were warnings during execution. Check the log for details. (4.41 s)
 (20/22) type_specific.io-github-autotest-libvirt.cpu.iothread.negative_test.iothread_quota_period_without_iothreads: WARN: Test passed but there were warnings during execution. Check the log for details. (50.50 s)
 (21/22) type_specific.io-github-autotest-libvirt.cpu.iothread.negative_test.disk_attach.no_iothread: WARN: Test passed but there were warnings during execution. Check the log for details. (93.57 s)
 (22/22) type_specific.io-github-autotest-libvirt.cpu.iothread.negative_test.disk_attach.delete_without_detach: WARN: Test passed but there were warnings during execution. Check the log for details. (93.17 s)
RESULTS    : PASS 0 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 22 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /var/lib/avocado/job-results/job-2023-11-13T20.07-eea4a32/results.html
JOB TIME   : 1153.80 s

**Comment by dzhengfy** [comment]
affected cases ' test result:
 (1/1) type_specific.io-github-autotest-libvirt.virsh.iothreadpin.normal_test.guest_on.id_option.option_live: WARN: Test passed but there were warnings during execution. Check the log for details. (51.01 s)
 (1/1) type_specific.io-github-autotest-libvirt.virsh.emulatorpin_mix.positive.change_with_disabled_cpuset: WARN: Test passed but there were warnings during execution. Check the log for details. (5.94 s)
 (1/1) type_specific.io-github-autotest-libvirt.virsh.iothreadinfo.normal_test.guest_on.id_option.option_live: WARN: Test passed but there were warnings during execution. Check the log for details. (51.06 s)
 (1/1) type_specific.io-github-autotest-libvirt.virsh.iothreaddel.normal_test.guest_on.id_option.option_live: WARN: Test passed but there were warnings during execution. Check the log for details. (105.29 s)
 (1/1) type_specific.io-github-autotest-libvirt.virsh.iothreadadd.normal_test.guest_on.id_option.option_live: WARN: Test passed but there were warnings during execution. Check the log for details. (50.42 s)

**Review by chloerh — APPROVED** [review]

File: libvirt/tests/src/cpu/iothread.py [hunk 1]
```diff
@@ -247,6 +254,73 @@ def exec_iothreadpin():
         if item not in xml_info.cputune.iothreadpins:
             test.fail("Unable to get {} from xml".format(item))
 
+    def _reassemble_iothread_config(mapping):
+        """
+        Reassemble iothreads using configuration
+
+        :param mapping: dict, mapping from command line option to object key
+        :ret: dict, the assembled iothreads
```
Comment (cliping) [inline libvirt/tests/src/cpu/iothread.py:262]: Maybe ret -> return

File: libvirt/tests/src/cpu/iothread.py [hunk 2]
```diff
@@ -298,14 +372,14 @@ def verify_poll():
             check_iothread_pool(UPDATE_IOTHREAD_POOL, exp_pool, True)
         logging.debug("Verify iothread poll: PASS")
 
-    def exec_iothreadset():
+    def exec_iothreadset(need_verify=True):
```
Comment (cliping) [inline libvirt/tests/src/cpu/iothread.py:375]: Missing need_verify docstring.

**Review by cliping — COMMENTED** [review]
Others LGTM

File: libvirt/tests/src/cpu/iothread.py (hunk 2 above)
Comment (dzhengfy) [reply libvirt/tests/src/cpu/iothread.py:375]: updated

File: libvirt/tests/src/cpu/iothread.py (hunk 1 above)
Comment (dzhengfy) [reply libvirt/tests/src/cpu/iothread.py]: updated

**Review by cliping — APPROVED** [review]

**Pull request merged**